code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,693 @@
+		examine_message += "<br>There are [egg_count] eggs able to be harvested! "
+
+	examine_message += "<br>"
+
+	//Report the number and types of live fish if there is water in the tank
+	if(fish_count == 0)
+		examine_message += "\The [src] doesn't contain any live fish. "
+	else
+		//Build a message reporting the types of fish
+		var/fish_num = fish_count
+		var/message = "You spot "
+		while(fish_num > 0)
+			if(fish_count > 1 && fish_num == 1)	//If there were at least 2 fish, and this is the last one, add "and" to the message
+				message += "and "
+			message += "a [fish_list[fish_num]]"
+			fish_num --
+			if(fish_num > 0)					//There's more fish, add a comma to the message
+				message +=", "
+		message +="."							//No more fish, end the message with a period
+		//Display the number of fish and previously constructed message
+		examine_message += "\The [src] contains [fish_count] live fish. [message] "
+
+	examine_message += "<br>"
+
+	//Report lid state for tanks and wall-tanks
+	if(has_lid)									//Only report if the tank actually has a lid
+		//Report lid state
+		if(lid_switch)
+			examine_message += "The lid is closed. "
+		else
+			examine_message += "The lid is open. "
+
+	examine_message += "<br>"
+
+	//Report if the tank is leaking/cracked
+	if(water_level > 0)							//Tank has water, so it's actually leaking
+		if(leaking == 1) examine_message += "\The [src] is leaking."
+		if(leaking == 2) examine_message += "\The [src] is leaking profusely!"
+	else										//No water, report the cracks instead
+		if(leaking == 1) examine_message += "\The [src] is cracked."
+		if(leaking == 2) examine_message += "\The [src] is nearly shattered!"
+
+
+	//Finally, report the full examine_message constructed from the above reports
+	to_chat(user, "[examine_message]")
+	return examine_message
+
+//////////////////////////////
+//		ATACK PROCS			//
+//////////////////////////////
+
+
+
+/obj/machinery/fishtank/attack_alien(mob/living/user as mob)
+	if(islarva(user)) return
+	attack_generic(user, 15)
+
+
+/obj/machinery/fishtank/attack_hand(mob/user as mob)
+
+	if(usr.a_intent == I_HURT)
+		playsound(get_turf(src), 'sound/effects/glassknock.ogg', 80, 1)
+		usr.visible_message("<span class='danger'>[usr.name] bangs against the [src.name]!</span>", \
+							"<span class='danger'>You bang against the [src.name]!</span>", \
+							"You hear a banging sound.")
+	else
+		playsound(src.loc, 'sound/effects/glassknock.ogg', 80, 1)
+		usr.visible_message("[usr.name] taps on the [src.name].", \
+							"You tap on the [src.name].", \
+							"You hear a knocking sound.")
+	return
+
+/obj/machinery/fishtank/proc/hit(var/damage, var/sound_effect = 1)
+	cur_health = max(0, cur_health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	check_health()
+
+/obj/machinery/fishtank/proc/attack_generic(mob/living/user as mob, damage = 0)	//used by attack_alien, attack_animal, and attack_slime
```

Comment:
as far as I can tell there is no reason for this proc to exist, because it is only called from one place, attack_alien